### PR TITLE
CI: Add XML check

### PR DESCRIPTION
**Pull request by jodh-intel**
Add a static analysis check for XML files.
  
Fixes: #1783.
    
Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: .ci/static-checks.sh [hunk 1]
```diff
@@ -838,6 +839,58 @@ static_check_vendor()
 	dep ensure -no-vendor -dry-run
 }
 
+static_check_xml()
+{
+	local all_xml
+	local files
+
+	all_xml=$(find . -name "*.xml" | grep -v "vendor/" | sort || true)
```
Comment (grahamwhaley) [inline .ci/static-checks.sh]: nitpick: don't know if `^vendor/` would be more accurate or not?

File: .ci/static-checks.sh [hunk 2]
```diff
@@ -838,6 +839,58 @@ static_check_vendor()
...
+
+		# Most XML documents are specified as XML 1.0 since, with the
+		# advent of XML 1.0 (Fifth Edition), XML 1.1 is "almost
+		# redundant" due to XML 1.0 providing the majority of XML 1.1
+		# features. xmllint doesn't support XML 1.1 seemingly for this
+		# reason, so the only check we can do is to (crudely) force
+		# the document to be an XML 1.0 one since XML 1.1 documents
+		# can mostly be represented as XML 1.0.
+		#
+		# This is only really required since Jenkins creates XML 1.1
```
Comment (grahamwhaley) [inline .ci/static-checks.sh]: Ah, so, this is with the (at least partial) intent of checking the Jenkins xml files... which, whilst good, if one of them is *not* valid XML, I'm not sure there is a huge amount we could do about it - the files are generated by Jenkins when we modify the GUI input - unclear if we would/could then go and hand-fix the xml and fold it back to the Jenkins server...

File: .ci/static-checks.sh [hunk 3]
```diff
@@ -838,6 +839,58 @@ static_check_vendor()
...
+		# documents.
+		contents=$(sed "s/xml version='1.1'/xml version='1.0'/g" "$file")
+
+		local ret
+
+		{ echo "$contents" | chronic xmllint -format -; ret=$?; } || true
```
Comment (grahamwhaley) [inline .ci/static-checks.sh]: nit: I think a `<<< $contents` is technically maybe more efficient than an `echo | ` :-)

File: .ci/static-checks.sh (hunk 1 above)
Comment (jodh-intel) [reply .ci/static-checks.sh]: I'll tal at this. We have a number of vendor-filtering points in the code, all subtly different... ;)

File: .ci/static-checks.sh (hunk 2 above)
Comment (jodh-intel) [reply .ci/static-checks.sh]: Although most of the XML files are Jenkins, there are other XML files in our repos. Regardless of who or what generated the files, we *really* do need to know if the files are valid or not. Particularly if you consider the fact that Jenkins is a core infrastructure tool and add a hypothetical disaster recovery scenario into the mix... :smile:

File: .ci/static-checks.sh (hunk 3 above)
Comment (jodh-intel) [reply .ci/static-checks.sh]: I guess this might save a few cycles - branch updated.

**Comment by jodh-intel** [comment]
@grahamwhaley - branch updated. This should do for now as a minimal change.

However, ultimately, I think the script needs a major refactor to simplify the logic.

**Pull request merged**